README.md

```diff
@@ -1,2 +1,186 @@
+Its goal is nothing more than to bring the scroll-behavior of full-page
+refreshes to an SPA (setting the scroll position to `(0, 0)` when a new "page" is
+navigated to).
+
+Sometimes, even when the URL path has changed, we don't want the screen position
+to reset.  This might be the case in an onboarding flow when navigating from
+`onboarding/step/1` to `onboarding/step/2`.  Navigating from `onboarding/step/2`
+to `/` however should reset the scroll position.
+
+If your app has `N` routes, you'd ostensibly have to declare how `N^2`
+transitions should be handled.  `react-simple-scroll` instead allows you to
+annotate your `react-router` route hierarchy with the "boundaries" of a page and
+handles all possible transitions for you.
+
+###### npm
+
+```bash
+npm install --save react-simple-scroll
+```
+
+###### yarn
+
+```bash
+yarn add react-simple-scroll
+```
+
+## Dependencies
+
+`react-simple-scroll` has no explicit dependencies, but will need you to provide
+three things:
+
+* React
+* `react-router`
+* An implementation of `isEqual`.  `isEqual` should accept two objects and
+  return `true` if their contents are deeply equal and `false` otherwise.
+  Lodash, underscore, et. al. ship with such a method.  It wasn't included in
+  this package assuming most users would already have an implementation
+  hanging around.
+
+## Quick Start
+
+To install `react-simple-scroll`, add it as a middleware to `<Router />`:
+
+```javascript
+import { Router, applyRouterMiddleware } from 'react-router';
+import { scrollMiddleware } from 'react-simple-scroll';
+import isEqual from 'lodash.isequal'
+
+<Router render={applyRouterMiddleware(scrollMiddleware({ isEqual }))>
+ {routes}
+</Router>
+```
+
+Next, annotate your `react-router` route hierarchy with the `scrollFrame` prop.
+A `scrollFrame` declares a "frame" within which we consider the page to have
+not transitioned.  Any route's frame is found by starting with itself and
+looking up the tree for the closest route which has the `srollFrame` property
+set to `true`. If the app transitions from route `A` to route `B` and they have
+the same scroll frame, the scroll position is not touched.  If they're
+different, the scroll position is reset to `(0, 0)`.
+
+```jsx
+const routes = (
+	<Route path="/">
+
+	  <Route path="foo" component={FooView} scrollFrame>
+	    <IndexRoute component={FooContainer} />
+	    <Route path="bar" component={BarContainer} />
+	    <Route path="baz" component={BazContainer} />
+	  </Route>
+
+	  <Route path="bloop" component={BloopView} scrollFrame>
+	    <IndexRoute component={BloopIndexContainer} />
+	    <Route scrollFrame>
+	      <Route path="bleep" component={BleepContainer} />
+	      <Route path="bleep/blap" component={BleepBlapContainer} />
+	      <Route path="bleep/blorp" component={BleepBlorpContainer} />
+	    </Route>
+	  </Route>
+
+	</Route>
+);
+```
+
+* `/foo/bar` to `/foo/baz` will not reset the scroll position
```

Comment:
This might be easier to understand as a table, e.g.:

|**From**|**To**|**Reset?**|
|---------|-----|-----|
`/foo/bar`|`/foo/baz`|No
`/foo`|`/foo/bar`|No
`/foo`|`/`|Yes